admin/services/access-arrangements.service.js

```diff
@@ -10,4 +16,65 @@ accessArrangementsService.getAccessArrangements = async () => {
   return accessArrangementsDataService.sqlFindAccessArrangements()
 }
 
-module.exports = accessArrangementsService
+/**
+ * Process access arrangements for single pupil
+ * @param {Object} requestData
+ * @param {Number} dfeNumber
+ * @param {Number} userId
+ * @returns {Object}
+ */
+accessArrangementsService.submit = async (requestData, dfeNumber, userId) => {
+  const validationError = accessArrangementsValidator.validate(requestData)
+  if (validationError.hasError()) {
+    throw validationError
+  }
+  const { pupilUrlSlug, accessArrangements: accessArrangementsCodes, questionReaderReason: questionReaderReasonCode } = requestData
+  let pupilAccessArrangements = R.clone(requestData)
+  const accessArrangements = await accessArrangementsDataService.sqlFindAccessArrangementsByCodes(accessArrangementsCodes)
+  let questionReaderReason
+  if (accessArrangements.length === 0) {
+    throw new Error('No access arrangements found')
+  }
+  pupilAccessArrangements['accessArrangements_ids'] = JSON.stringify(accessArrangements.map(aa => aa.id))
+  const pupil = await pupilDataService.sqlFindOneBySlugAndSchool(pupilUrlSlug, dfeNumber)
+  if (!pupil) {
+    throw new Error('Pupil url slug does not match a pupil record')
+  }
+  pupilAccessArrangements['pupil_id'] = pupil.id
+  pupilAccessArrangements['recordedBy_user_id'] = userId
+  if (!pupilAccessArrangements.accessArrangements.includes('ITA')) {
+    pupilAccessArrangements = R.omit(['inputAssistanceInformation'], pupilAccessArrangements)
```

Comment:
IMHO this is just weird.  You are using `omit` to get a new data structure, and then proceed to re-assign the reference to it.   This is incongruous; if we are using immutable objects then please be consistent.

Reply:
@jon-shipley this is now removed and the pattern has changed as per our conversation